src/js/node/stream.js

```diff
@@ -2624,6 +2627,24 @@ var require_readable = __commonJS({
       $debug("about to maybereadmore");
       maybeReadMore(stream, state);
     }
+    function onEofChunk(stream, state) {
```

Comment:
In JS, property accesses are slower than variable declarations - so lets only do these once

```suggestion
    function onEofChunk(stream, state) {
     if (state.ended) return;
     const decoder = state.decoder;
      if (decoder) {
        const chunk = decoder.end();
        const chunkLength = chunk?.length;
        if (chunkLength) {
          state.buffer.push(chunk);
          state.length += state.objectMode ? 1 : chunkLength;
        }
      }
      state.ended = true;
      if (state.sync) {
        emitReadable(stream, state);
      } else {
        state.needReadable = false;
        state.emittedReadable = true;
        _emitReadable(stream, state);
      }
    }
```